docs/how-to-setup-wsl.md

```diff
@@ -1,240 +1,130 @@
+   > [!NOTE]
+   >
+   > While you may use other non-debian based distros, they all come with their own gotchas and are beyond the scope of this guide.
 
-6) Return to the GitHub SSH setup to test your connection
+2. Update the dependencies for the OS
 
-### ALTERNATIVE 3 - Last Resort
+   ```console
+   sudo apt update
+   sudo apt upgrade -y
 
-1) Install build-essential, fakeroot and dpkg-dev using the following command:
+   # cleanup
+   sudo apt autoremove -y
+   ```
 
-```sh
-sudo apt-get install build-essential fakeroot dpkg-dev
-```
-
-2) Create a directory named git-rectify in the home folder using the following command:
-
-```sh
-mkdir ~/git-rectify
-```
-
-3) Change directory to the `get-rectify` directory and get the git source files:
-
-```sh
-cd ~/git-rectify
-```
-
-4) This command creates a copy of your `sources.list` file, for safe keeping.
-
-```sh
-sudo cp /etc/apt/sources.list /etc/apt/sources.list~
-```
-
-> [!NOTE]
-> If you encounter any errors with the directory not existing, or the file being copied to already existing, follow this step:
-
- - Enter the directory
-
-```sh
-cd ~/git-rectify/etc/apt
-```
-
- - Open your file explorer
+## Set up Git
 
-```sh
-explorer.exe .
-```
-
- - Manually create a copy of the `sources.list` file, and save it with a different name.
- - Change directory back to `git-rectify`
+Git comes pre-installed with Ubuntu 18.04, verify that your Git version with `git --version`.
 
-```sh
-cd ../..
+```output
+~
+❯ git --version
+git version 2.25.1
 ```
 
-Continue from these steps:
-
-5) Now, you need to uncomment the sources in the ../ file
-
-```sh
-sudo sed -Ei 's/^# deb-src /deb-src /' /etc/apt/sources.list
-```
+(Optional but recommended) You can now proceed to [setting up your ssh keys](https://help.github.com/articles/generating-an-ssh-key) with GitHub.
 
-6) Get the source files from the list:
+## Installing a Code Editor
 
-```sh
-apt-get source git
-```
+We highly recommend installing [Visual Studio Code](https://code.visualstudio.com) on Windows 10. It has great support for WSL and automatically installs all the necessary extentions on your WSL distro.
 
-7) Install all the git dependencies:
+Essentially, you will edit and store your code on Ubuntu-18.04 with VS Code installed on Windows.
 
-```sh
-sudo apt-get build-dep git
-```
+## Installing Docker Desktop
 
-8) Install libcurl with all development files:
+**Docker Desktop for Windows** allows you to install and run database and services like MongoDB, NGINX, etc. This is useful to avoid common pitfalls with installing MongoDB or other services directly on Windows or WSL2.
 
-```sh
-sudo apt-get install libcurl4-openssl-dev
-```
+Follow the instructuction on the [official documentation](https://docs.docker.com/docker-for-windows/install) and install Docker Desktop for your Windows distribution.
 
-9) Unpack all the source packages using the following command:
+There are some minimum hardware requirements for the best experience.
```

Comment:
It might be best to put this health warning earlier so people don't get here and become annoyed.  If you know some rough specs, I'd include them.

Reply:
You are right. I do include that at the very top of the file, it's easy to spot on the rendered version of the document.

But I should probably link that here again?

Reply:
Sure, I'd either remove it entirely or put something like

> If you encounter problems with this setup, please double-check that your system meets the requirements `<link>`